src/state/application/updater.ts

```diff
@@ -35,11 +35,27 @@ function useBlockWarningTimer() {
   const timeout = useRef<NodeJS.Timeout>()
   const isWindowVisible = useIsWindowVisible()
   const [msSinceLastBlock, setMsSinceLastBlock] = useState(0)
-  const currentBlock = useBlockNumber()
+  const blockTimestamp = useCurrentBlockTimestamp()
 
   useEffect(() => {
-    setMsSinceLastBlock(0)
-  }, [currentBlock])
+    if (blockTimestamp && chainId) {
+      const waitMsBeforeWarning =
+        (chainId ? CHAIN_INFO[chainId]?.blockWaitMsBeforeWarning : DEFAULT_MS_BEFORE_WARNING) ??
+        DEFAULT_MS_BEFORE_WARNING
+      const now = Date.now()
+      setMsSinceLastBlock(Math.floor(now - blockTimestamp.mul(1000).toNumber()))
+      if (msSinceLastBlock > waitMsBeforeWarning) {
+        if (!chainConnectivityWarningActive) {
+          dispatch(setChainConnectivityWarning({ warn: true }))
+        }
+      } else {
+        if (chainConnectivityWarningActive) {
+          dispatch(setChainConnectivityWarning({ warn: false }))
+        }
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
smell. also this doesn't need state anymore, it's just whether the ms since last block is greater than wait ms before warning

Reply:
🤦‍♂️